telemetry/specs/telemetry_report.mdk

```diff
@@ -424,47 +439,108 @@ The Destination Port is user configurable. The expectation is that the same
-      |  Ver  |Length |NProt| RepMdBits | Reserved  |D|Q|F|   hw_id   |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                           Switch id                           |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                        Sequence Number                        |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                       Ingress Timestamp                       |
+      |  Ver  |   hw_id   |           Sequence Number                 |
       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                  Variable Optional Metadata                   |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-`
+      |                            Node id                            |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+      |RepType| InType| Report Length |   MD Length   |D|Q|F|I|  Rsvd |  |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  R
+      |            RepMdBits          |      Domain Specific ID       |  E
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  P
+      |            DSMdBits           |          DSMdstatus           |  O                        
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  R
+      |          Variable Optional Baseline & DS Metadata             |  T                   
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  |
+      |    Truncated Packet or  Additional DS Extension Data or TLV   |  |  
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+      
+* Ver (4b): Version
+  : This specification defines **version 2.0**.
+
+* hw_id (6b): Hardware ID
+  : Identifies the hardware subsystem within the network device that generated
+    this report. For example, in a chassis with multiple linecards this could
+    identify a specific linecard, or a subsystem within a linecard. The hw_id
+    is unique within the scope of a Switch id.
+
+* Sequence Number (22b): Sequence Number
+  : Reflects the sequence of reports from a specific combination of
+    (Node id, hw_id) to a particular telemetry report destination. This
+    can be used to detect loss of telemetry reports before they reach their
+    intended destination.
 
-* Ver: Version
-  : This specification defines **version 1**.
+* Node ID (32b): Node ID
+  : The unique ID of a network device (such as routers, switches, NICs etc.) This is generally administratively assigned. Node IDs
+    must be unique within a management domain.
 
-* Length
-  : Indicates the length of the telemetry report header in multiples of
-    4 octets, including the variable optional metadata.
+* RepType (4b): Report Type
+  : Type of Telemetry report
+    * 0: INT v2.0 or later.
+    * 1: IOAM
+    * 2 – 15: Reserved
```

Comment:
I might have missed a discussion on RepType. It feels we need some clarifications on this field. 

1. I think we exclude INT v1.0 or 0.5 to be used with Report sepc 2.0. Can we clarify that?
1. IOAM version doesn't matter because the embedded IOAM header will have its own version field?
1. Is RepType the type of in-band telemetry embedded in the Truncated Packet? or does it matter even when there is no Truncated Packet?

Reply:
Will add text with respect to INT v1.0 and v0.5 to clarify use of Report Spec v2.0
I would say it is still TBD. I can change it IOAM v(TBD).
RepType identifies if the Telemetry report is for INT v2.0 or for any other Telemetry.  It does not matter if a Truncated packet is there or not.

Reply:
Bytes 4-11 after RepType are intentionally aligned in format with bytes 4-11 of INT-MX / INT-MD headers, so they can just be copied over when necessary.

We are hoping to take the same approach with IOAM, changing the format of bytes 4-11 to align with the IOAM Trace format, including all the IOAM bit definitions. This will make it easier to support IOAM Direct Export without having to remap instruction bits.

Reply:
I see, that's neat.
Then it sounds like each RepType value should map to a version of each unique INT or IOAM header.
is 4b RepType large enough for that? 
Or the mapping of RepType value to INT/IOAM version can change when Telemetry Report spec version changes?

Reply:
IOAM with Namespace extensions, similar to INT Domain Specific Extensions don't really map directly the way INT header bytes 4-11 are defined.  Namespace specific Trace Type and Namespace Specific Flags are not part of IOAM Header.  It is going to much harder to get the Namespace Extensions fields as part of default IOAM header.

Reply:
I think mapping of RepType value to INT/IOAM version can change with Telemetry Report Spec version changes.

Reply:
@jklr I see 3 major enhancements in Telemetry Report 2.0 compared to Telemetry Report 1.0:

1. Support for domain specific extensions
2. Some new RepMdBit values
3. Coalescing of multiple reports in one packet

If supporting INT 1.0 or INT 0.5, there are no domain specific extensions. It looks like one RepMdBit value is missing in Telemetry Report 1.0 compared to INT 1.0, which is Level 2 Ingress Port ID + Egress Port ID (4 bytes each). I guess we did not want to go to a mix of 4 and 8 bytes per RepMdBit. Unless we want to cut a quick Telemetry Report 1.1 before merging this PR, that will be a drawback.

If supporting INT 1.0 or 0.5 and coalescing or level 2 port IDs are required, using the "INT 2.0" RepType looks like it would work. Perhaps we should just call it "INT" rather than "INT 2.0", even though it was designed with INT 2.0 in mind.

Reply:
@rsivakolundu Since IOAM does not have Namespace specific Trace Type and Namespace Specific Flags, where would the values come from?

If they are needed, it would have to be done through Domain Specific Extensions or equivalent, in a TLV if the truncated packet fragment is also required.

Reply:
It is your call.

Reply:
It is your call for Telemetry Report 1.1.

Reply:
As far as Namespace Extensions, the current thinking is to trigger of Namespace ID and have an extension to the IOAM header to carry Namespace Specific Instructions and Flags.

Reply:
@rsivakolundu The IOAM data draft is in between last calls. Are people thinking about making major changes just before the supposed final last call?
If not, when/how would the format be extended?

For IOAM direct export, given the way the current proposal determines whether flow ID and sequence number are present based on length, it seems rather unfriendly to future extensions.

Reply:
For Telemetry Report 2.0, the proposal for now is to reserve a RepType codepoint for IOAM, but the format that goes along with that is not yet defined. I guess we would cut Telemetry Report 2.1 when the IOAM formats get finalized.

Reply:
Makes sense to cut Telemetry Report 2.1 when IOAM formats are finalized.